README.md

```diff
@@ -120,6 +120,40 @@ main().then(() => {
 
 </details>
 
+<details>
+  <summary>AbortController</summary>
+
+```javascript
+import { RequestManager, Client, HTTPTransport, AbortError } from "@open-rpc/client-js";
+
+const transport = new HTTPTransport("http://localhost:3333");
+const requestManager = new RequestManager([transport]);
+const client = new Client(requestManager);
+
+const main = async () => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+
+  setTimeout(() => {
+    controller.abort();
+  }, 100);
+
+  try {
+    await client.request({ method: "long_running_method", params: [], timeout: 5000, signal });
```

Comment:
```suggestion
    await client.request({ method: "long_running_method", params: [] }, {timeout: 5000, signal });
```

I think we should leave the jsonrpc request arg as only JSON-RPC, and have an argument for the options.

Reply:
I agree here. for better background , we have context which might be an argument that we can pass in here as a Context Object. We can then use this around the project for various things. I imagine the Context argument type would be ClientContext where all fields are optional a Partial type.

Reply:
I would just call it `Options`, `Context` feels like the wrong name for this.

Reply:
Pushed some changes introducing an `Options` interface to carry timeout/signal.